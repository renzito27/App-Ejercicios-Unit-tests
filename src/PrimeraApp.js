import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
//const PrimeraApp = ({saludo = 'Hola Mundo' }) => {}
const PrimeraApp = ({saludo, subtitulo}) => {

    //const saludo = 'Hola Renziz!';
    // const hi = {
        // nombre : 'Renzo',
        // edad : 37
    // };  renderizar objetos

    return (
        <Fragment>
            <h1>{saludo}</h1>
            {/* <pre>{JSON.stringify(hi,null,3)}</pre> */} 
            {/* renderizar objetos */}
            <p>{subtitulo}</p>
        </Fragment>

     );
}
    
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired

}   

PrimeraApp.defaultProps = {
    subtitulo : 'Soy el subt√≠tulo'
}
    
export default PrimeraApp;