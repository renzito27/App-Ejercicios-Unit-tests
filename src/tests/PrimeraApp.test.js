import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
//import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';


describe ('Pruebas en componente <primeraApp/>', () => {

   /*  test ('debe de mostrar el mensaje "Hola soy Goku"',() => {

       const saludo = "Hola, soy Goku" ;
       const {getByText} = render(<PrimeraApp saludo = { saludo }/>);
       expect(getByText(saludo)).toBeInTheDocument();
    }); */

    test('debe de mostrar <PrimeraApp/> correctamente', () => {
        const saludo = "Hola, soy Goku";
        const wrapper = shallow(<PrimeraApp saludo ={ saludo}/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        const saludo = "Hola, soy Goku";
        const subTitulo = 'Soy un subtitulo';
        const wrapper = shallow(
            <PrimeraApp 
                saludo ={ saludo}
                subtitulo = { subTitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);
        
        
    });


});