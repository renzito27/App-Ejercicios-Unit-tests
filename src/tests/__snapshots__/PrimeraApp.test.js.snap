// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <primeraApp/> debe de mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy el subt√≠tulo
  </p>
</Fragment>
`;
